### PR TITLE
Support disabling default response rendering

### LINKED ISSUE #2979: Option to turn off default 200 response generation in protoc-gen-openapiv2
## 🚀 Feature

I was wondering if it is feasible to have a similar option to disableDefaultErrors to turn off the default generation of the 200 responses? Has this been considered and dismissed?

Some examples:

- I have a post request where I only want to return 201 Created (or error codes)
- I have a delete request where I only want to return 204 No Content (or error codes)
- I have a head request where I only want to return 204 No Content

This would marry nicely with the ability to inject a httpResponseModifier in the mux

### PR DESCRIPTION
**Pull request by krak3n**
Fixes #2979

# Changes

Addes a new `disable_default_responses` flag (default `false`) which disables the rendering of the default responses. This allows users to define different API responses for cases for where `200` is not the default response code from an existing API, for example it maybe `201`.

The side effect of this is all responses would need to be defined when this option is set `true`.

### CODE REVIEW

File: protoc-gen-openapiv2/internal/genopenapi/template_test.go [hunk 1]
```diff
@@ -3316,6 +3316,17 @@ func TestApplyTemplateProtobufAny(t *testing.T) {
...
+		{
+			// we have a protobufAny in a message but with automatic rendering of responses disabled
+			name: "protobufAny_referenced_in_message_with_default_responses_disabled",
+			args: args{
+				msgContainsAny: true,
+				regConfig: func(reg *descriptor.Registry) {
+					reg.SetDisableDefaultResponses(true)
+				},
+			},
+			wantNumDefinitions: 4,
```
Comment (krak3n) [inline protoc-gen-openapiv2/internal/genopenapi/template_test.go:3328]: Not exactly sure what this tests but it's here 😄

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
LGTM! One thing, do you think you could add a small section to our docs about this new capability too? They live in https://github.com/grpc-ecosystem/grpc-gateway/blob/master/docs/docs/mapping/customizing_openapi_output.md. Thanks!

**Comment by krak3n** [comment]
@johanbrandhorst add example output to the documentation 👍

File: docs/docs/mapping/customizing_openapi_output.md [hunk 2]
```diff
@@ -631,14 +631,105 @@ info:
...
 paths:
   /api/hello:
     get:
       operationId: EchoService_Hello
-...
+```
+
+### Disable default responses
+
+By default a 200 OK response is rendered for each service operation. But it is possible to disable this and explicitly define your service's responses, using the `disable_default_responses` option. Allowed values are: `true`, `false`.
```
Comment (johanbrandhorst) [inline docs/docs/mapping/customizing_openapi_output.md:645]: Could you just add a note somewhere to say that this does _not_ change the behavior of the gateway itself, and must be coupled with a custom ForwardResponseWriter that adjusts the response to be an accurate reflection of the API behavior? Maybe link to the docs section that talks about changing the response code.

**Review by johanbrandhorst — COMMENTED** [review]
Thanks for the docs, just a small idea.

File: docs/docs/mapping/customizing_openapi_output.md (hunk 2 above)
Comment (krak3n) [reply docs/docs/mapping/customizing_openapi_output.md:645]: Added a note @johanbrandhorst

**Review by johanbrandhorst — APPROVED** [review]

**Comment by johanbrandhorst** [comment]
Thanks for your contribution!

**Comment by zshainsky** [comment]
Hi all what would it take to add this to the bazel `protoc_gen_openapiv2` command? Would that be just a simple update to the `defs.bzl` file?

**Comment by johanbrandhorst** [comment]
Yep, that should be it. We should've added it when we added this flag, oops 😬

**Comment by zshainsky** [comment]
Thanks @johanbrandhorst want me to open an issue for this?

**Comment by johanbrandhorst** [comment]
Seems like @rajukrishnamurthy beat you to it in #4123.

**Pull request merged**